### PR TITLE
TE-3.1: Moving interface-ref check for next-hop under if block to avoid indirect nh validation

### PR DESCRIPTION
**Pull request by trathod1**
The interface-ref validation is not applicable for indirect next-hop. Moving interface-ref check based on discussion in - https://github.com/openconfig/featureprofiles/issues/964

"This code is a Contribution to the OpenConfig Feature Profiles project ("Work") made under the Google Software Grant and Corporate Contributor License Agreement ("CLA") and governed by the Apache License 2.0. No other rights or licenses in or to any of Nokia's intellectual property are granted for any other purpose. This code is provided on an "as is" basis without any warranties of any kind."

### CODE REVIEW

File: feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 1]
```diff
@@ -350,11 +350,11 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 		}
 		nh := gnmi.Get(t, args.dut, gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().NextHop(nhIndexInst).State())
 		if got, want := nh.GetIpAddress(), atePort2.IPv4; got != want {
+			if nh.GetInterfaceRef().GetInterface() == "" {
```
Comment (xw-g) [inline feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go:353]: if got != want, we don't even need to look at the interface ref, right?

File: feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 2]
```diff
@@ -316,12 +316,12 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 			// for devices that return the nexthop without resolving it recursively. For a->b->c the device returns b
 			// note that b->c is validated in the next code block
 			if got != ateIndirectNH {
+				if nh.GetInterfaceRef().GetInterface() == "" {
```
Comment (xw-g) [inline feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go:319]: This whole block of code (starting from `if got := nh.GetIpAddress(); got != atePort2.IPv4`) lacks of readability and confuses ppl.  Let's take this chance to improve it. 

How about something like below? 
```
if got := nh.GetIpAddress(); got != atePort2.IPv4 && got != ateIndirectNH {
	t.Errorf("next-hop is incorrect: got %v, want %v or %v ", got, ateIndirectNH, atePort2.IPv4)
}
```

File: feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 3]
```diff
@@ -316,12 +316,12 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
...
+					t.Errorf("next-hop interface-ref/interface not found")
+				}
 				t.Errorf("next-hop is incorrect: got %v, want %v or %v ", got, ateIndirectNH, atePort2.IPv4)
 			}
 		}
-		if nh.GetInterfaceRef().GetInterface() == "" {
```
Comment (xw-g) [inline feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go:322]: It only checks if the interface is empty, but not if the ref interface is correct. I also don't think it needed here for now. 

I think we can just remove this code, and let https://github.com/openconfig/featureprofiles/pull/1132 handles it later.

File: feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 4]
```diff
@@ -350,11 +350,11 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 		}
 		nh := gnmi.Get(t, args.dut, gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().NextHop(nhIndexInst).State())
 		if got, want := nh.GetIpAddress(), atePort2.IPv4; got != want {
+			if nh.GetInterfaceRef().GetInterface() == "" {
+				t.Errorf("next-hop interface-ref/interface not found")
+			}
 			t.Errorf("next-hop is incorrect: got %v, want %v", got, want)
 		}
-		if nh.GetInterfaceRef().GetInterface() == "" {
```
Comment (xw-g) [inline feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go:355]: same as above.

**Review by xw-g — COMMENTED** [review]
Do we need o update OTG too? Thanks.

File: feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go (hunk 3 above)
Comment (trathod1) [reply feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go:322]: Sure, will remove this.

File: feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go (hunk 4 above)
Comment (trathod1) [reply feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go:355]: Sure, will remove this.

**Comment by trathod1** [comment]
> Do we need o update OTG too? Thanks.

Yes, similar changes are required for OTG as well. I will commit it here.

**Comment by MarcCharlebois** [comment]
@xw-g have you had a chance to review the changes?

**Comment by xw-g** [comment]
b/272057375 is the internal presubmit effort.

**Review by xw-g — APPROVED** [review]

**Pull request merged**